### PR TITLE
Travis CI

### PR DESCRIPTION
**Pull request by marktwtn**
CI: Setup Travis CI for unit testing
test: Return -1 exit code when unit test fails

### CODE REVIEW

**Comment by marktwtn** [comment]
Currently we have some failed unit testing.
They will be fixed in the future.

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  - arm64
+
+os:
+  - linux
+
+compiler:
+  - gcc
+
+before_install:
+  - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then sudo apt-get install -y build-essential; fi
```
Comment (jserv) [inline .travis.yml]: Is it necessary to validate Linux?

Comment (marktwtn) [reply .travis.yml]: No, I'll remove it.

Comment (marktwtn) [reply .travis.yml]: Done.

**Pull request merged**